ssl/ssl3.h

```diff
@@ -627,6 +627,9 @@ typedef struct ssl3_state_st {
      */
     unsigned char *alpn_selected;
     unsigned alpn_selected_len;
+    unsigned char *alpn_proposed;
```

Comment:
This is in the ssl3_state_st structure; which, as far as I know, is fairly well hidden (although can be referenced via s->s3). As long as I put it at the end (which may not be the case here), it ought to be OK? (Yes, I'm aware of OpenSSH and 1.0.2b/c.) I can put it into s->cert if absolutely necessary.